src/plugins/contracts/api.js

```diff
@@ -202,8 +202,30 @@ function cancelContract(web3) {
   }
 }
 
+function purchaseContract(web3, cloneFactory, lumerin) {
+  return async (params) => {
+    const { walletId, contractId, url, privateKey, price } = params;
+    const sendOptions = { from: walletId, gas: 1_000_000}
+
+    const account = web3.eth.accounts.privateKeyToAccount(privateKey)
+    web3.eth.accounts.wallet.create(0).add(account)
+
+    await web3.eth.getTransactionCount(walletId, 'pending')
```

Comment:
what is this function used for?

Reply:
Unnecessary call, that was part of what existed logic on wallet